### PR TITLE
[HTTP3] Correctly handle server closing a control stream

### PR DESCRIPTION
**Pull request by rzikm**
This PR makes sure we behave according to the spec when server closes/aborts either its or clients control stream (i.e. close the connection with `H3_CLOSED_CRITICAL_STREAM` error)

### CODE REVIEW

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs [hunk 1]
```diff
@@ -368,6 +369,16 @@ private async Task SendSettingsAsync()
             try
             {
                 _clientControl = await _connection!.OpenOutboundStreamAsync(QuicStreamType.Unidirectional).ConfigureAwait(false);
+
+                // Server MUST NOT abort our control stream, setup a continuation which will react accordingly
+                _ = _clientControl.WritesClosed.ContinueWith(t =>
+                {
+                    if (t.Exception?.InnerExceptions.FirstOrDefault() is QuicException ex && ex.QuicError == QuicError.StreamAborted)
```
Comment (ManickaP) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs:376]:
```suggestion
                    if (t.Exception?.InnerException is QuicException ex && ex.QuicError == QuicError.StreamAborted)
```

**Review by ManickaP — APPROVED** [review]
LGTM, thanks.

**Review by CarnaViire — APPROVED** [review]

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@
 using System.Runtime.Versioning;
 using System.Net.Quic;
 using System.IO;
+using System.Linq;
```
Comment (ManickaP) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs:10]:
```suggestion
```

Comment (rzikm) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs:10]: oops, I missed this one and merged it, I will remove it when I next touch HTTP3

**Pull request merged**